packages/strapi-plugin-upload/middlewares/upload/index.js

```diff
@@ -23,5 +26,41 @@ module.exports = strapi => ({
     });
 
     strapi.router.get('/uploads/(.*)', range, koaStatic(staticDir, { defer: true }));
+
+    strapi.router.get('/upload/proxy', async ctx => {
+      try {
+        const url = new URL(ctx.query.url);
+
+        if (!['http:', 'https:'].includes(url.protocol)) {
```

Comment:
How about using regex?

Reply:
Why would you use a regex to check if a value is equal to some constants ?

Reply:
> How about using regex?

Why would you use a regex to check if a value is equal to some constants ?